tools/integration-test/src/tests/query_packet.rs

```diff
@@ -0,0 +1,149 @@
+use ibc_relayer::chain::counterparty::{channel_on_destination, pending_packet_summary};
+use ibc_relayer::link::{Link, LinkParameters};
+
+use ibc_test_framework::prelude::*;
+use ibc_test_framework::relayer::channel::query_identified_channel_end;
+use ibc_test_framework::relayer::connection::query_identified_connection_end;
+use ibc_test_framework::util::random::random_u64_range;
+
+#[test]
+fn test_query_packet_pending() -> Result<(), Error> {
+    run_binary_channel_test(&QueryPacketPendingTest)
+}
+
+pub struct QueryPacketPendingTest;
+
+impl TestOverrides for QueryPacketPendingTest {
+    fn modify_relayer_config(&self, config: &mut Config) {
```

Comment:
Just want to clarify: for the purposes of this test, is it fine that pending packets are relayed once the relayer has been started? That's what will happen if `config.mode.tx_confirmation` is `true`, but I don't know what the default value of `tx_confirmation` is. 

If it's `false`, then pending packets won't be cleared due to the interaction between `tx_confirmation` and `clear_interval`. If it's `true`, then they will be cleared on account of the interaction. 

If it doesn't matter that pending packets are cleared after the relayer has been started, then I think it's also fine to ignore this interaction.

Reply:
The test never runs the supervisor, packets are relayed step by step by invoking methods on a `Link`.
This is to get the packet summaries into an expected state without dependency on timing.